src/apps/external_executor/external_executor.cpp

```diff
@@ -129,6 +124,170 @@ namespace externalexecutor
       return nullptr;
     }
 
+    std::unordered_map<std::string, MapStrategyPtr> map_index_strategies;
+
+    void install_index_service(ccfapp::AbstractNodeContext& node_context)
+    {
+      auto executor_auth_policy = std::make_shared<ExecutorAuthPolicy>();
+      ccf::AuthnPolicies executor_only{executor_auth_policy};
+
+      // registers the index and will store the streamptr
+      auto install_and_subscribe_index =
+        [this, &node_context](
+          ccf::endpoints::CommandEndpointContext& ctx,
+          externalexecutor::protobuf::IndexInstall&& payload,
+          ccf::grpc::StreamPtr<externalexecutor::protobuf::IndexWork>&&
+            out_stream) -> ccf::grpc::GrpcAdapterStreamingResponse {
+        std::string strategy = payload.strategy_name();
+
+        auto it = map_index_strategies.find(strategy);
+        if (it != map_index_strategies.end())
+        {
+          return ccf::grpc::make_error(
+            GRPC_STATUS_ALREADY_EXISTS,
+            fmt::format("Strategy {} already exists", strategy));
+        }
+        auto executor_id = get_caller_executor_id(ctx);
+
+        auto ds = payload.data_structure();
+        // Mark default ds as MAP
+        IndexDataStructure data_structure = MAP;
+
+        if (ds == externalexecutor::protobuf::IndexInstall::PREFIX_TREE)
+        {
+          data_structure = PREFIX_TREE;
+        }
+
+        // signal to the indexer that it is now subscribed
+        externalexecutor::protobuf::IndexWork work;
+        work.mutable_subscribed();
+        out_stream->stream_msg(work);
+
+        std::shared_ptr<ExecutorIndex> map_index =
+          std::make_shared<ExecutorIndex>(
+            payload.map_name(),
+            strategy,
+            data_structure,
+            executor_id,
+            ctx,
+            node_context,
+            std::move(out_stream));
+
+        node_context.get_indexing_strategies().install_strategy(map_index);
+
+        // store the index pointer into the strategies
+        map_index_strategies[strategy] = map_index;
```

Comment:
Comment for future, but not for this PR: if we separate registration from data structure construction, then we won't have a name to use as the key here. I think we can instead use the executor's cert as the key, to store this indexing strategy instance.